### PR TITLE
[LS] Fix IndexOutOfBoundException in STModify API

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
$subject. 

Fixes https://github.com/wso2/product-ballerina-integrator/issues/135

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by kanushka — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java [hunk 1]
```diff
@@ -243,32 +242,54 @@ private static boolean importExist(UnusedSymbolsVisitor unusedSymbolsVisitor, AS
...
     }
+
+    public static TextRange getRange(ASTModification modification, TextDocument oldTextDocument, String textEdit) {
+        LinePosition startLinePos = LinePosition.from(modification.getStartLine(), modification.getStartColumn());
+        LinePosition endLinePos = LinePosition.from(modification.getEndLine(), modification.getEndColumn());
+
+        int startOffset;
+        try {
+            startOffset = oldTextDocument.textPositionFrom(startLinePos);
+            int endOffset = oldTextDocument.textPositionFrom(endLinePos);
```
Comment (nipunayf) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java:279]: As we As we discussed, we need two try-catch cases:

One for the startPosition (s → eof and diff → 0)

And one for the endPosition (s → s and diff → eof - s)

Comment (NipunaRanasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java:279]: since 2 scenarios throw different exceptions, I guess its okay to add a single try block with multiple catch blocks? its already added in L290

Comment (NipunaRanasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java:279]: anyway didnt implement the fallback logic for case 2 (refer to the TODO) considering it as an edge case and, as we need to test and verify the fallback logic before adding it. WDYT?

Comment (nipunayf) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java:279]: That's not the case I am mentioning.

There are two lines that can cause the IndexOutOfBounds exception: one for the start and one for the end. In your case, you're assigning EOF to the start even when it's in the correct range.

For example, if EOF = 50 and s = 45:
Your implementation sets s = 50 and diff = 0
Ideally, it should be s = 45 and diff = 5

There should be separate try-catch blocks for each statement, catching the same IndexOutOfBound exception.

Comment (NipunaRanasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java]: well that makes sense. even though that case didnt occur in config variable scenarios, it still can happen. 
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/43976/commits/8bf556fd080525c738949db25ee7870591dc431d

**Review by nipunayf — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]
